src/Adapter/Order/OrderProductUpdater.php

```diff
@@ -0,0 +1,561 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Order;
+
+use Address;
+use Cart;
+use CartRule;
+use Configuration;
+use Context;
+use Country;
+use Currency;
+use Customer;
+use Db;
+use Language;
+use Order;
+use OrderCarrier;
+use OrderCartRule;
+use OrderDetail;
+use OrderInvoice;
+use Pack;
+use PrestaShop\PrestaShop\Adapter\ContextStateManager;
+use PrestaShop\PrestaShop\Adapter\Order\Refund\OrderProductRemover;
+use PrestaShop\PrestaShop\Adapter\StockManager;
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\OrderException;
+use PrestaShop\PrestaShop\Core\Localization\CLDR\ComputingPrecision;
+use Product;
+use StockAvailable;
+use StockManagerFactory;
+use StockMvt;
+use Tools;
+use Validate;
+use Warehouse;
+
+/**
+ * Increase or decrease quantity of an order's product.
+ * Recalculate cart rules, order's prices and shipping infos.
+ */
+class OrderProductUpdater
```

Comment:
Reading this description and the content of the method, it seems more logical to call the method `updateQuantity`

Reply:
I renamed the service to OrderProductQuantityUpdater ;)